SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -65,27 +108,49 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 				"Actor: %s. Strategy: %s"),
 		   *Actor->GetName(), *NetDriver->LoadBalanceStrategy->GetName());
 
-	Worker_ComponentId ActorInterestComponentId = ClassInfoManager->ComputeActorInterestComponentId(Actor);
+	AuthorityDelegationMap DelegationMap;
+	DelegationMap.Add(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthoritativeServerPartitionId);
+	DelegationMap.Add(SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID, AuthoritativeClientPartitionId);
 
-	for (auto& SubobjectInfoPair : Info.SubobjectInfo)
+	// Add debugging utilities, if we are not compiling a shipping build
+#if !UE_BUILD_SHIPPING
+	if (NetDriver->SpatialDebugger != nullptr)
 	{
-		const FClassInfo& SubobjectInfo = SubobjectInfoPair.Value.Get();
+		check(NetDriver->VirtualWorkerTranslator != nullptr);
 
-		// Static subobjects aren't guaranteed to exist on actor instances, check they are present before adding to delegation component
-		TWeakObjectPtr<UObject> Subobject = PackageMap->GetObjectFromUnrealObjectRef(FUnrealObjectRef(EntityId, SubobjectInfoPair.Key));
-		if (!Subobject.IsValid())
-		{
-			continue;
-		}
+		const PhysicalWorkerName* PhysicalWorkerName =
+			NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(IntendedVirtualWorkerId);
+		FColor InvalidServerTintColor = NetDriver->SpatialDebugger->InvalidServerTintColor;
+		FColor IntentColor =
+			PhysicalWorkerName != nullptr ? SpatialGDK::GetColorForWorkerName(*PhysicalWorkerName) : InvalidServerTintColor;
+
+		const bool bIsLocked = NetDriver->LockingPolicy->IsLocked(Actor);
+
+		SpatialDebugging DebuggingInfo(SpatialConstants::INVALID_VIRTUAL_WORKER_ID, InvalidServerTintColor, IntendedVirtualWorkerId,
+									   IntentColor, bIsLocked);
+		ComponentDatas.Add(DebuggingInfo.CreateComponentData());
 	}
+#endif // !UE_BUILD_SHIPPING
 
-	// We want to have a stably named ref if this is an Actor placed in the world.
-	// We use this to indicate if a new Actor should be created, or to link a pre-existing Actor when receiving an AddEntityOp.
-	// Previously, IsFullNameStableForNetworking was used but this was only true if bNetLoadOnClient=true.
-	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server
-	// crashes and restarts.
-	TSchemaOption<FUnrealObjectRef> StablyNamedObjectRef;
-	TSchemaOption<bool> bNetStartup;
+	// Add actual load balancing components
+	ComponentDatas.Add(NetOwningClientWorker(AuthoritativeClientPartitionId).CreateComponentData());
+	ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(IntendedVirtualWorkerId));
```

Comment:
Nit: this uses a different style from other components

Reply:
Yeah, for some reason it was like that even before. I think I will make it consistent.

Reply:
Made consistent